examples/plot_pp_study.py

```diff
@@ -0,0 +1,362 @@
+#   accuracy of the model is 95%. Does that mean the model is good to go?
+#   Can there still be issues despite what seems like a properly working
+#   model? Tradeoffs between accuracy and fairness?
+#
+# * What does it mean that the last bin has the highest selection rate?
+#   Given the sociotechnical context, what would this imply in terms of
+#   the societal impact and outcome?
+#
+# * What would a ‘mitigation’ mean in this scenario? Assuming no
+#   constraints on the mitigations offered in Fairlearn modules, what type
+#   of disparity would we be looking to reduce? (Explored in the section
+#   to follow)
+#
+# Another important question to consider: should the described problem
+# (policing) be framed as an ML task at all?
+#
+# **Exploration: What if we attempt to apply Fairlearn's mitigation functionality?**
+#
+#
+# *Quickstart - Mitigating Disparity*
+#
+#
+# If we observe disparities between groups we may want to create a new
+# model while specifying an appropriate fairness constraint. Note that the
+# choice of fairness constraints is crucial for the resulting model, and
+# varies based on application context. If selection rate is highly
+# relevant for fairness in this contrived example, we can attempt to
+# mitigate the observed disparity using the corresponding fairness
+# constraint called Demographic Parity. In real world applications we need
+# to be mindful of the sociotechnical context when making such decisions.
+# The Exponentiated Gradient mitigation technique used fits the provided
+# classifier using Demographic Parity as the objective, leading to a
+# vastly reduced difference in selection rate.
+#
+# *Hypothesis*
+#
+#
+# In this sociotechnical context, the disparity is inherent in the data
+# itself, and thus an attempt at mitigation will fail.
+#
+np.random.seed(0)  # set seed for consistent results with ExponentiatedGradient
+constraint = DemographicParity()
+classifier = DecisionTreeClassifier(min_samples_leaf=10, max_depth=4)
+mitigator = ExponentiatedGradient(classifier, constraint)
+mitigator.fit(data_without_crimerate, y_true, sensitive_features=lstat_binned)
+y_pred_mitigated = mitigator.predict(data_without_crimerate)
+
+sr_mitigated = MetricFrame(selection_rate, y_true, y_pred_mitigated, sensitive_features=lstat_binned)
+print(sr_mitigated.overall)
+print(sr_mitigated.by_group)
+
+as_mitigated = MetricFrame(accuracy_score, y_true, y_pred_mitigated, sensitive_features=lstat_binned)
+print(as_mitigated.overall)
+print(as_mitigated.by_group)
+# %%
+#
+# By constraining selection rate disparity across LSTAT groups, we notice
+# a significant difference in the accuracy for the 3rd and 4th bins:
+#
+# **Before Mitigation - 95%+ accuracy on both bins**
+#
+#
+# **After Mitigation - 74% and 42% respectively**
+#
+#
+# The significant drop in accuracy indicates that the underlying issue may
+# not be at the model level but rather a reflection of the data (and
```

Comment:
I agree, but why does that drop in accuracy suggest that? Will that be obvious to the reader?